pkg/sources/s3/s3.go

```diff
@@ -198,53 +199,62 @@ func (s *Source) getBucketsToScan(client *s3.S3) ([]string, error) {
 	return bucketsToScan, nil
 }
 
-func (s *Source) scanBuckets(ctx context.Context, client *s3.S3, role string, bucketsToScan []string, chunksChan chan *sources.Chunk) {
-	objectCount := uint64(0)
+func (s *Source) scanBuckets(
+	ctx context.Context,
+	client *s3.S3,
+	role string,
+	bucketsToScan []string,
+	chunksChan chan *sources.Chunk,
+) {
+	var objectCount uint64
 
-	logger := s.log
 	if role != "" {
-		logger = logger.WithValues("roleArn", role)
+		ctx = context.WithValue(ctx, "role", role)
 	}
 
 	for i, bucket := range bucketsToScan {
-		logger := logger.WithValues("bucket", bucket)
+		ctx := context.WithValue(ctx, "bucket", bucket)
 
 		if common.IsDone(ctx) {
 			return
 		}
 
-		s.SetProgressComplete(i, len(bucketsToScan), fmt.Sprintf("Bucket: %s", bucket), "")
-		logger.V(3).Info("Scanning bucket")
+		s.SetProgressComplete(i, len(bucketsToScan), "Bucket: "+bucket, "")
```

Comment:
I have no opinion about your other stylistic changes, but this one took me a second to understand. Do you have a strong objection to retaining `fmt.Sprintf` here?

Reply:
Happy to change it back. I'll also consider disabling perfsprintf from my linter.